### PR TITLE
PIP-902-trim-adapters-illumina

### PR DESCRIPTION
**Pull request by ottojolanki**

### CODE REVIEW

File: tests/integration/test_illumina.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - "-f"
+      - "-1 P5"
+      - "-2 P7"
+      - "--threads=1"
+      - "trim.R1.fastq.gz"
+      - "trim.R2.fastq.gz"
+      - "&> trimstats.txt"
+  files:
+    - path: "test-output/trim.R1.fastq.gz"
+    - path: "test-output/trim.R2.fastq.gz"
```
Comment (keenangraham) [inline tests/integration/test_illumina.yml]: What about trimstats.txt? And md5?

File: wdl/tasks/illumina.wdl [hunk 2]
```diff
@@ -0,0 +1,46 @@
+version 1.0
+
+
+import "../structs/resources.wdl"
+import "../structs/fastq.wdl"
```
Comment (keenangraham) [inline wdl/tasks/illumina.wdl]: Do we care about import order?

File: wdl/tasks/illumina.wdl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        String read2_out_filename = "trim.R2.fastq.gz"
+        String trimstats_out_filename = "trimstats.txt"
+    }
+
+    command {
+        trim-adapters-illumina \
+            -f ~{adapters} \
+            -1 ~{adapter1} \
+            -2 ~{adapter2} \
+            --threads=~{resources.cpu} \
```
Comment (keenangraham) [inline wdl/tasks/illumina.wdl]: You didn't want to wrap these in TrimAdapterParams? Also here I let threads be independently specified even though is should probably be the same as cpu number: https://github.com/ENCODE-DCC/dnase-seq-pipeline/blob/363a4eca990d310ed3e5c2bc72b20302b60442a7/wdl/tasks/bwa.wdl#L55 Should we pick one way or the other?

File: tests/integration/test_illumina.yml (hunk 1 above)
Comment (ottojolanki) [reply tests/integration/test_illumina.yml]: I have requested, they have not provided.

File: wdl/tasks/illumina.wdl (hunk 2 above)
Comment (ottojolanki) [reply wdl/tasks/illumina.wdl]: Yes, I think we should. Will fix.

File: tests/integration/test_illumina.yml (hunk 1 above)
Comment (ottojolanki) [reply tests/integration/test_illumina.yml]: forgot about trimstats.txt thanks.

File: wdl/tasks/illumina.wdl (hunk 3 above)
Comment (ottojolanki) [reply wdl/tasks/illumina.wdl]: Case by case basis agreed.

**Pull request merged**